sdk/workflow/workflow.go

```diff
@@ -27,12 +27,20 @@ func New(instance *instancesdk.Instance, workflowDB database.WorkflowDB) *Workfl
 func (w *Workflow) Create(wf *workflow.Workflow) (*workflow.Workflow, error) {
 	wf.Hash = hash.Dump(wf)
 
-	if _, err := w.instance.Get(wf.Trigger.InstanceHash); err != nil {
-		return nil, err
-	}
 	for _, node := range wf.Nodes {
-		if _, err := w.instance.Get(node.InstanceHash); err != nil {
-			return nil, err
+		switch n := node.(type) {
```

Comment:
If the `node` is an interface the method `GetInstanceHash` could be added and used here instead of switch. Otherwise:

```
    switch n := node.(type) {
    case workflow.Result:
      instanceHash = n.InstanceHash
    case workflow.Event:
      instanceHash = n.InstanceHash
    case workflow.Task:
      instanceHash = n.InstanceHash
    }
    if _, err := w.instance.Get(instanceHash); err != nil { 
      return nil, err
    }
```

Reply:
`instanceHash` is specific to some types of node, not all of them, mapping and filter doesn't have this.

Maybe I can use this `case workflow.Result, workflow.Event...`

Reply:
> instanceHash is specific to some types of node, not all of them, mapping and filter doesn't have this.

Ok. So in order to have this method, they need to return nil on `GetInstanceHash` if we want. such behavior. 

> Maybe I can use this case workflow.Result, workflow.Event...

Nope, it won't work here.

Reply:
But instance hash is not in all the types of node so id doesn't make sense to have this for each node. We could try to have another interface that is `nodeWithInstanceHash` or something like that but that's definitely overkilled in my opinion

Reply:
So switch stays :)